src/query/sensor/processors/sensor.processor.ts

```diff
@@ -131,46 +140,52 @@ export class SensorProcessor {
       sensorData = {...sensorData, typeDetails: event.typeDetails};
     }
 
-    Sensor.updateOne({_id: event.sensorId}, sensorData, (err) => {
-      if (err) {
-        this.logError(event);
-      }
-    });
+    const sensor: ISensor = await Sensor.findByIdAndUpdate(
+      event.sensorId,
+      sensorData,
+      this.errorCallback,
+    ).exec();
+
+    return sensor;
   }
 
-  async processDeleted(event: SensorDeleted): Promise<void> {
-    Sensor.deleteOne({_id: event.sensorId}, (err) => {
-      if (err) {
-        this.logger.error('Error while deleting projection.');
-      }
-    });
+  async processDeleted(event: SensorDeleted): Promise<ISensor> {
+    const sensor: ISensor = await Sensor.findByIdAndDelete(
+      event.sensorId,
+    ).exec();
+
+    return sensor;
   }
 
-  async processActivated(event: SensorActivated): Promise<void> {
+  async processActivated(event: SensorActivated): Promise<ISensor> {
     const sensorData = {
       active: true,
     };
 
-    Sensor.updateOne({_id: event.sensorId}, sensorData, (err) => {
-      if (err) {
-        this.logError(event);
-      }
-    });
+    const sensor: ISensor = await Sensor.findByIdAndUpdate(
```

Comment:
Dit stuk lijkt wel telkens hetzelfde, niet? Kan deze niet naar een aparte method? _(en dan direct uit deze methods en in slechts één method call vanuit de `process(event)` method?)_

Reply:
Dat laatste waarschijnlijk niet, zie ik nu ... 🤔

Reply:
Dat lijkt inderdaad zo, maar er komen variaties voor 😉 Er wordt voornamelijk `findByIdAndUpdate` gedaan, maar `findOneAndUpdate` en `findByIdAndDelete` komen ook voor, één aparte method gaat daardoor niet lukken. Toch gaat het wel mooier zijn om het wat minder DRY te maken, zal daarom de db interacties even in aparte functies te zetten 👍